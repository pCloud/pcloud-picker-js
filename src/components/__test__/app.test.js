import * as React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";

import { App, Modal } from "..";

const appProps = {
  mode: "select",
  clientId: "lh4j5JDeeCB",
  buttonText: "PcloudButton",
  fileUrl: "https://i.ytimg.com/vi/YCaGYUIfdy4/maxresdefault.jpg",
  isFolderSelectionOnly: false,
  onSelect: jest.fn(),
  onClose: jest.fn()
};

describe("<App />", () => {
  let app, appInstance;

  beforeEach(() => {
    app = shallow(<App {...appProps} />);
    appInstance = app.instance();
  });

  it("renders component correctly", () => {
    expect(app).toMatchSnapshot();
  });

  describe("<Wrapper />", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = app.find("App__Wrapper");
    });

    it("renders component correctly", () => {
      expect(wrapper.length).toBeGreaterThan(0);
    });

    it("contains everything else that gets rendered", () => {
      const wrapperDiv = wrapper.first();
      expect(wrapper.children()).toEqual(app.children());
    });
  });

  describe("<Modal />", () => {
    let modal;

    beforeEach(() => {
      modal = app.find("Modal");
    });

    it("redners component correctly", () => {
      expect(modal.length).toBe(1);
    });

    it("sets component props porps", () => {
      expect(modal.prop("show")).toEqual(app.state("isModalOpened"));
      expect(modal.prop("closeModal")).toEqual(appInstance.closeModal);
    });
  });

  describe("<Picker />", () => {
    let picker;
    beforeEach(() => {
      app.setState({ isAuthenticated: true });
      picker = app.find("Picker");
    });

    it("redners component correctly when isAuthenticated is set", () => {
      expect(app.find("Picker").length).toBe(1);
    });

    it("set component props correctly", () => {
      expect(picker.prop("isFolderSelectionOnly")).toEqual(
        appInstance.props.isFolderSelectionOnly
      );
      expect(picker.prop("getFolderContent")).toEqual(
        appInstance.getFolderContent
      );
      expect(picker.prop("onPick")).toEqual(appInstance.onPick);
      expect(picker.prop("onCancel")).toEqual(appInstance.onCancel);
    });
  });

  describe("<DefaultButton />", () => {
    let defaultButton;

    beforeEach(() => {
      defaultButton = app.find("App__DefaultButton");
    });

    it("should render component correctly", () => {
      expect(defaultButton.length).toBe(1);
    });

    it("sets component props correctly", () => {
      expect(defaultButton.prop("onClick")).toEqual(appInstance.getToken);
    });

    it("renders button text correctly", () => {
      expect(defaultButton.children().text()).toBe(
        appInstance.props.buttonText
      );
    });

    it("should get user token and open modal", () => {
      expect(app.state("isAuthenticated")).toBeFalsy();
      expect(app.state("isModalOpened")).toBeFalsy();

      defaultButton.simulate("click");

      expect(app.state("isAuthenticated")).toBeTruthy();
      expect(app.state("isModalOpened")).toBeTruthy();
    });
  });
});
