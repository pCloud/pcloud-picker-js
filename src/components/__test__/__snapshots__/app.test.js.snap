// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render <App /> renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    buttonText="PcloudButton"
    clientId="lh4j5JDeeCB"
    fileUrl="https://i.ytimg.com/vi/YCaGYUIfdy4/maxresdefault.jpg"
    isFolderSelectionOnly={false}
    mode="select"
    onClose={[MockFunction]}
    onSelect={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Modal
          closeModal={[Function]}
          show={false}
        />,
        <App__DefaultButton
          onClick={[Function]}
        >
          PcloudButton
        </App__DefaultButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "closeModal": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "PcloudButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "PcloudButton",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Modal
            closeModal={[Function]}
            show={false}
          />,
          <App__DefaultButton
            onClick={[Function]}
          >
            PcloudButton
          </App__DefaultButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "closeModal": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "PcloudButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "PcloudButton",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<App /> renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    buttonText="PcloudButton"
    clientId="lh4j5JDeeCB"
    fileUrl="https://i.ytimg.com/vi/YCaGYUIfdy4/maxresdefault.jpg"
    isFolderSelectionOnly={false}
    mode="select"
    onClose={[MockFunction]}
    onSelect={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Modal
          closeModal={[Function]}
          show={false}
        />,
        <App__DefaultButton
          onClick={[Function]}
        >
          PcloudButton
        </App__DefaultButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "closeModal": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "PcloudButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "PcloudButton",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Modal
            closeModal={[Function]}
            show={false}
          />,
          <App__DefaultButton
            onClick={[Function]}
          >
            PcloudButton
          </App__DefaultButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "closeModal": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "PcloudButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "PcloudButton",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
